### PR TITLE
packable: add a struct-level verify_with

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+fn verify<const VERIFY: bool>(&value: &u64, _: &()) -> Result<(), PickyError> {
+    if !VERIFY || value == 42 {
+        Ok(())
+    } else {
+        Err(PickyError(value as u8))
```
Comment (Thoralf-M) [inline packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs:28]: Why is the value accepted as u64 if it needs to be an u8 at the end?

Comment (thibault-martinez) [reply packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs:28]: Copy paste error

Comment (thibault-martinez) [reply packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs:28]: Ah no, not copy paste error. This is a failing test, it's supposed to be a wrong type. I've put u64 but I could have put Vec, it' just there to make it fail.

Comment (Thoralf-M) [reply packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs:28]: So providing a wrong type will fail to compile already, right?

Comment (thibault-martinez) [reply packable/packable-derive-test/tests/fail/invalid_field_type_verify_with_struct.rs:28]: Yes

**Review by Thoralf-M — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

File: packable/packable-derive-test/Cargo.toml
Comment (Alex6323) [inline packable/packable-derive-test/Cargo.toml:1]: also bump its version?

Comment (thibault-martinez) [reply packable/packable-derive-test/Cargo.toml:1]: No point, this is not released

File: packable/packable/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.9.0 - 2023-11-14
+
+- Struct-level `verify_with` attribute;
+
```
Comment (Alex6323) [inline packable/packable/CHANGELOG.md:25]: Sub-Header (### Added) missing?

**Review by Alex6323 — APPROVED** [review]

File: packable/packable-derive-test/Cargo.toml
Comment (Alex6323) [reply packable/packable-derive-test/Cargo.toml:1]: (it has been bumped before, that's why I am asking)

**Review by kwek20 — APPROVED** [review]

File: packable/packable-derive-test/Cargo.toml
Comment (thibault-martinez) [reply packable/packable-derive-test/Cargo.toml:1]: I've reset it to 0.0.0 so it's less confusing

File: packable/packable/CHANGELOG.md (hunk 2 above)
Comment (thibault-martinez) [reply packable/packable/CHANGELOG.md]: Done

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**